libs/pages/application/src/lib/ui/page-settings-advanced/page-settings-advanced.tsx

```diff
@@ -0,0 +1,186 @@
+  keys?: string[]
+  defaultAdvancedSettings?: ApplicationAdvancedSettings
+  advancedSettings?: ApplicationAdvancedSettings
+  loading: LoadingStatus
+  onSubmit?: () => void
+  discardChanges: () => void
+}
+
+export function PageSettingsAdvanced(props: PageSettingsAdvancedProps) {
+  const { control, formState } = useFormContext()
+  const [showOverriddenOnly, toggleShowOverriddenOnly] = useState(false)
+
+  let table: EditionTableRow[] = [
+    {
+      cells: [
+        {
+          content: 'Settings',
+        },
+        {
+          content: 'Default Value',
+        },
+        {
+          content: <span className="px-3">Value</span>,
+          className: '!p-1',
+        },
+      ],
+      className: 'font-medium',
+    },
+  ]
+
+  const tableBody: EditionTableRow[] = props.keys
+    ? props.keys?.map((key) => {
+        return {
+          className: `${
+            showOverriddenOnly &&
+            props.defaultAdvancedSettings &&
+            props.advancedSettings &&
+            props.defaultAdvancedSettings[key as keyof ApplicationAdvancedSettings]?.toString() ===
+              props.advancedSettings[key as keyof ApplicationAdvancedSettings]?.toString()
+              ? 'hidden'
+              : ''
+          }`,
+          cells: [
+            {
+              content: (
+                <Tooltip content={key}>
+                  <div>{key}</div>
+                </Tooltip>
+              ),
+              className: 'font-medium vis',
+            },
+            {
+              content: (
+                <>
+                  <Tooltip
+                    content={
+                      (props.defaultAdvancedSettings &&
+                        props.defaultAdvancedSettings[key as keyof ApplicationAdvancedSettings]?.toString()) ||
+                      ''
+                    }
+                  >
+                    <div className="inline whitespace-nowrap overflow-hidden text-ellipsis">
+                      {props.defaultAdvancedSettings &&
+                        props.defaultAdvancedSettings[key as keyof ApplicationAdvancedSettings]?.toString()}
+                    </div>
+                  </Tooltip>
+                  <CopyToClipboard
+                    className="ml-2 text-text-300 invisible group-hover:visible"
+                    content={
+                      (props.defaultAdvancedSettings &&
+                        props.defaultAdvancedSettings[key as keyof ApplicationAdvancedSettings]?.toString()) ||
+                      ''
+                    }
+                  />
+                </>
+              ),
+              className: 'group',
+            },
+            {
+              className: '!p-1',
+              content: (
+                <Controller
+                  name={key}
+                  control={control}
+                  rules={{
+                    required: 'Please enter a value.',
+                  }}
+                  defaultValue=""
```

Comment:
defaultValue necessary too?

Reply:
I think due to the strange behavior of this specific case where backend send us key on this form `"bla.blo.blu"` react-hook-form is doing a weird computation things and I could not make it work without error without specifying the defaultValue here. But we won't need this `defaultValue` in normal cases where the key are... normal.